File: branches/Medusa/Source/Fantasy.Jobs/Management/JobThread.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;

namespace Fantasy.Jobs.Management
{
    internal class JobThread
    {
        public JobThread(JobMetaData job, Thread thread, bool isResume)
        {
            this.Thread = thread;
            this.Job = job;
            this.IsResume = isResume;
            this.ExitEvent = new ManualResetEvent(false);
        }

        public Thread Thread { get; private set; }


        public JobMetaData Job { get; private set; }


        public ManualResetEvent ExitEvent { get; private set; }


        public bool IsResume { get; private set; }

        public IJobEngine Engine { get; set; }
    }
}

File: Fantasy.Jobs/src/fantasy/jobs/management/JobThread.java
package fantasy.jobs.management;

public class JobThread
{
	public JobThread(JobMetaData job, Thread thread, boolean isResume)
	{
		this.setThread(thread);
		this.setJob(job);
		this.setIsResume(isResume);
		this.setExitEvent(new ManualResetEvent(false));
	}

	private Thread privateThread;
	public final Thread getThread()
	{
		return privateThread;
	}
	private void setThread(Thread value)
	{
		privateThread = value;
	}


	private JobMetaData privateJob;
	public final JobMetaData getJob()
	{
		return privateJob;
	}
	private void setJob(JobMetaData value)
	{
		privateJob = value;
	}


	private ManualResetEvent privateExitEvent;
	public final ManualResetEvent getExitEvent()
	{
		return privateExitEvent;
	}
	private void setExitEvent(ManualResetEvent value)
	{
		privateExitEvent = value;
	}


	private boolean privateIsResume;
	public final boolean getIsResume()
	{
		return privateIsResume;
	}
	private void setIsResume(boolean value)
	{
		privateIsResume = value;
	}

	private IJobEngine privateEngine;
	public final IJobEngine getEngine()
	{
		return privateEngine;
	}
	public final void setEngine(IJobEngine value)
	{
		privateEngine = value;
	}
}